lib/pdf_template.rb

```diff
@@ -38,4 +38,27 @@ def render(locals)
 
     tmp
   end
+
+  def rebase_urls(html)
+    doc = Nokogiri::HTML.fragment(html)
+
+    if ENV["SERVER_HOST"]
+      url_base = %(#{ENV["SERVER_PROTOCOL"]}://#{ENV["SERVER_HOST"]})
+      url_base << %(:#{ENV["SERVER_PORT"]}) if ENV["SERVER_PORT"]
+
+      doc.css("a, img").each do |a|
+        if a["href"] && a["href"] !~ /^https?:\/\//
+          path = a["href"].sub(/^\//, "")
+          a["href"] = %(#{url_base}/#{path})
+        end
+
+        if a["src"] && a["src"] !~ /^https?:\/\//
+          path = a["src"].sub(/^\//, "")
+          a["src"] = %(#{url_base}/#{path})
```

Comment:
fancy!

Reply:
... uhh I'll fix this. I can't remember how the unusual string syntax got here.